iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/ShortBackupRetentionCheck.java

```diff
@@ -69,7 +90,75 @@ private void checkBackupRetentionDatabaseAccounts(ContextualResource resource) {
     }
   }
 
-  private static Predicate<Expression> isNumericValue(Predicate<Float> predicate) {
+  private void checkBackupRetentionRecoveryServicesVaults(ContextualResource resource) {
+    retrieveRetentionPolicyObjects(resource).forEach(retentionPolicyObject -> {
+      ContextualObject retentionDurationObject = retrieveRetentionDurationObject(retentionPolicyObject);
+      Double durationInDays = computeRetentionInDays(retentionDurationObject);
+      if (durationInDays != null && durationInDays < retentionPeriodInDays) {
+        retentionDurationObject.property("count").report(RETENTION_PERIOD_TOO_SHORT_MESSAGE, retentionDurationObject.property("durationType").toSecondary("Duration type"));
+      }
+    });
+  }
+
+  private static Stream<ContextualObject> retrieveRetentionPolicyObjects(ContextualResource resource) {
```

Comment:
```suggestion
  private static Stream<ContextualObject> retrieveRetentionPolicy(ContextualResource resource) {
```
For me, it would be enough without the `Object` suffix. The return type is fine. Same for `retrieveRetentionDuration`